sdk/storage/inc/blobs/blob_options.hpp

```diff
@@ -191,178 +190,178 @@ namespace Azure { namespace Storage { namespace Blobs {
+    Azure::Core::Nullable<std::string> ContentCRC64;
     BlobHttpHeaders Properties;
     std::map<std::string, std::string> Metadata;
-    AccessTier Tier = AccessTier::Unknown;
-    std::string IfModifiedSince;
-    std::string IfUnmodifiedSince;
-    std::string IfMatch;
-    std::string IfNoneMatch;
+    Azure::Core::Nullable<AccessTier> Tier;
+    Azure::Core::Nullable<std::string> IfModifiedSince;
+    Azure::Core::Nullable<std::string> IfUnmodifiedSince;
+    Azure::Core::Nullable<std::string> IfMatch;
+    Azure::Core::Nullable<std::string> IfNoneMatch;
   };
 
   struct StageBlockOptions
   {
     Azure::Core::Context Context;
-    std::string ContentMD5;
-    std::string ContentCRC64;
+    Azure::Core::Nullable<std::string> ContentMD5;
+    Azure::Core::Nullable<std::string> ContentCRC64;
   };
 
   struct StageBlockFromUriOptions
   {
     Azure::Core::Context Context;
-    uint64_t SourceOffset = std::numeric_limits<uint64_t>::max();
-    uint64_t SourceLength = 0;
-    std::string ContentMD5;
-    std::string ContentCRC64;
-    std::string LeaseId;
-    std::string SourceIfModifiedSince;
-    std::string SourceIfUnmodifiedSince;
-    std::string SourceIfMatch;
-    std::string SourceIfNoneMatch;
+    Azure::Core::Nullable<int64_t> SourceOffset;
+    Azure::Core::Nullable<int64_t> SourceLength;
+    Azure::Core::Nullable<std::string> ContentMD5;
+    Azure::Core::Nullable<std::string> ContentCRC64;
+    Azure::Core::Nullable<std::string> LeaseId;
+    Azure::Core::Nullable<std::string> SourceIfModifiedSince;
+    Azure::Core::Nullable<std::string> SourceIfUnmodifiedSince;
+    Azure::Core::Nullable<std::string> SourceIfMatch;
+    Azure::Core::Nullable<std::string> SourceIfNoneMatch;
   };
 
   struct CommitBlockListOptions
   {
     Azure::Core::Context Context;
     BlobHttpHeaders Properties;
     std::map<std::string, std::string> Metadata;
-    AccessTier Tier = AccessTier::Unknown;
-    std::string IfModifiedSince;
-    std::string IfUnmodifiedSince;
-    std::string IfMatch;
-    std::string IfNoneMatch;
+    Azure::Core::Nullable<AccessTier> Tier;
+    Azure::Core::Nullable<std::string> IfModifiedSince;
+    Azure::Core::Nullable<std::string> IfUnmodifiedSince;
+    Azure::Core::Nullable<std::string> IfMatch;
+    Azure::Core::Nullable<std::string> IfNoneMatch;
   };
 
   struct GetBlockListOptions
   {
     Azure::Core::Context Context;
-    BlockListTypeOption ListType = BlockListTypeOption::All;
-    std::string IfModifiedSince;
-    std::string IfUnmodifiedSince;
-    std::string IfMatch;
-    std::string IfNoneMatch;
+    Azure::Core::Nullable<BlockListTypeOption> ListType;
+    Azure::Core::Nullable<std::string> IfModifiedSince;
+    Azure::Core::Nullable<std::string> IfUnmodifiedSince;
+    Azure::Core::Nullable<std::string> IfMatch;
+    Azure::Core::Nullable<std::string> IfNoneMatch;
   };
 
   struct CreateAppendBlobOptions
   {
     Azure::Core::Context Context;
     BlobHttpHeaders Properties;
```

Comment:
Why not `Nullable`?

Reply:
it's already nullable

Reply:
Let me get this straight, you want to make everything (literally) in options nullable, don't you?

Reply:
Yes, I believe that's what we discussed and decided with Jeffrey.

Reply:
As far as I can remember, what we've decided is that, we don't use a specific value (0, MIN or MAX) to represent this parameter isn't specified.

Have we also decided that everything in options must be Nullable? At least I didn't find this conclusion in Jan. 11 meeting minutes.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/232